src/server/worker/driver/driver_unix.go

```diff
@@ -107,17 +107,22 @@ func (d *driver) linkData(inputs []*common.Input, dir string) error {
 		return err
 	}
 
+	// sometimes for group inputs, this part may get run multiple times for the same file
+	seen := make(map[string]bool)
 	for _, input := range inputs {
 		if input.S3 {
 			continue // S3 data is not downloaded
 		}
 		if input.Name == "" {
 			return errors.New("input does not have a name")
 		}
-		src := filepath.Join(dir, input.Name)
-		dst := filepath.Join(d.InputDir(), input.Name)
-		if err := os.Symlink(src, dst); err != nil {
-			return err
+		if _, ok := seen[input.Name]; !ok {
+			seen[input.Name] = true
+			src := filepath.Join(dir, input.Name)
+			dst := filepath.Join(d.InputDir(), input.Name)
+			if err := os.Symlink(src, dst); err != nil {
+				return err
+			}
```

Comment:
I wonder if it's possible to do a merge step on `input` rather than patching it here. I suspect there are some other places that assume there's only one input per directory and it might be easier to maintain the invariant than to fix it at this step.